enable N return in characters()

Al-Murphy opened the pull request:
Hey Jacob, amazing work on this extremely useful package! I've started moving over to using it as a base starting piint for any work I'm doing and so if I think of any other use cases I'll try submitting them as PRs. For this one, I ran into an issue using `characters()` which may be intentional but, it can't seem to handle missing bases i.e. 'N' (tangermeme v0.2.3):

```
from tangermeme.utils import characters,one_hot_encode
a = one_hot_encode("ACGTN")
print(a)
tensor([[[1, 0, 0, 0, 0],
         [0, 1, 0, 0, 0],
         [0, 0, 1, 0, 0],
         [0, 0, 0, 1, 0]]], dtype=torch.int8)
characters(a)
---------------------------------------------------------------------------
File ~/anaconda3/envs/g-cads/lib/python3.12/site-packages/tangermeme/utils.py:178, in characters(pwm, alphabet, force)
    [176](https://vscode-remote+ssh-002dremote-002b146-002d169-002d8-002d78-002edsi-002eic-002eac-002euk.vscode-resource.vscode-cdn.net/shared/aemurphy/G-CADS/~/anaconda3/envs/g-cads/lib/python3.12/site-packages/tangermeme/utils.py:176) pwm_ismax = pwm == pwm.max(dim=0, keepdims=True).values
    [177](https://vscode-remote+ssh-002dremote-002b146-002d169-002d8-002d78-002edsi-002eic-002eac-002euk.vscode-resource.vscode-cdn.net/shared/aemurphy/G-CADS/~/anaconda3/envs/g-cads/lib/python3.12/site-packages/tangermeme/utils.py:177) if pwm_ismax.sum(axis=0).max() > 1 and force == False:
--> [178](https://vscode-remote+ssh-002dremote-002b146-002d169-002d8-002d78-002edsi-002eic-002eac-002euk.vscode-resource.vscode-cdn.net/shared/aemurphy/G-CADS/~/anaconda3/envs/g-cads/lib/python3.12/site-packages/tangermeme/utils.py:178) 	raise ValueError("At least one position in the PWM has multiple " +
    [179](https://vscode-remote+ssh-002dremote-002b146-002d169-002d8-002d78-002edsi-002eic-002eac-002euk.vscode-resource.vscode-cdn.net/shared/aemurphy/G-CADS/~/anaconda3/envs/g-cads/lib/python3.12/site-packages/tangermeme/utils.py:179) 		"letters with the same probability.")
    [181](https://vscode-remote+ssh-002dremote-002b146-002d169-002d8-002d78-002edsi-002eic-002eac-002euk.vscode-resource.vscode-cdn.net/shared/aemurphy/G-CADS/~/anaconda3/envs/g-cads/lib/python3.12/site-packages/tangermeme/utils.py:181) alphabet = numpy.array(alphabet)
    [182](https://vscode-remote+ssh-002dremote-002b146-002d169-002d8-002d78-002edsi-002eic-002eac-002euk.vscode-resource.vscode-cdn.net/shared/aemurphy/G-CADS/~/anaconda3/envs/g-cads/lib/python3.12/site-packages/tangermeme/utils.py:182) if isinstance(pwm, torch.Tensor):

ValueError: At least one position in the PWM has multiple letters with the same probability.
```

I believe having a parameter (False by default seems sensible) to enable conversion to N's?

I've implemented a fix on this but let me know if I'm missing something important on including this?

Cheers,
Alan.

Al-Murphy commented:
Edit - Just a small addition to also enable N values in the `_validate_input()` from tangermeme.utils

jmschrei commented:
Ah, good catch. Thanks!

jmschrei commented:
Would you mind adding this change to `docs/whats_new.rst`?

jmschrei commented:
It looks like the change has caused a fair number of errors. Can you look into those?

Al-Murphy commented:
Corrected now, apologies! I've added the change to the list of changes for v0.2.4 in `whats_new.rst`, didn't know if you wanted me to bump the version?

Al-Murphy commented:
Sorry, ready now!

Al-Murphy commented:
Will create a clean PR with all additions added
